src/descriptor/policy.rs

```diff
@@ -787,3 +787,344 @@ impl ExtractPolicy for Descriptor<DescriptorPublicKey> {
+             && m == &2
+             && items.len() == 3
+             && conditions.get(&vec![0,1]).unwrap().iter().next().unwrap().csv.is_none()
+             && conditions.get(&vec![0,2]).unwrap().iter().next().unwrap().csv == Some(sequence)
+             && conditions.get(&vec![1,2]).unwrap().iter().next().unwrap().csv == Some(sequence)
+            )
+        );
+    }
+
+    // - mixed timelocks should fail
+
+    #[test]
+    fn test_extract_policy_for_wsh_mixed_timelocks() {
+        let (prvkey0, _pubkey0, _fingerprint0) = setup_keys(TPRV0_STR);
+        let locktime_threshold = 500000000; // if less than this means block number, else block time in seconds
+        let locktime_blocks = 100;
+        let locktime_seconds = locktime_blocks + locktime_threshold;
+        let desc = descriptor!(sh (and_v (+v pk prvkey0), (and_v (+v after locktime_seconds), (after locktime_blocks)))).unwrap();
+        let (wallet_desc, keymap) = desc.to_wallet_descriptor(Network::Testnet).unwrap();
+        let signers_container = Arc::new(SignersContainer::from(keymap));
+        let policy = wallet_desc
+            .extract_policy(signers_container)
+            .unwrap()
+            .unwrap();
+
+        println!("desc policy = {:?}", policy); // TODO remove
+
+        // TODO how should this fail with mixed timelocks?
+    }
+
+    // - multiple timelocks of the same type should be correctly merged together
+
+    #[test]
+    fn test_extract_policy_for_multiple_same_timelocks() {
+        let (prvkey0, _pubkey0, _fingerprint0) = setup_keys(TPRV0_STR);
+        let locktime_blocks0 = 100;
+        let locktime_blocks1 = 200;
+        let desc = descriptor!(sh (and_v (+v pk prvkey0), (and_v (+v after locktime_blocks0), (after locktime_blocks1)))).unwrap();
+        let (wallet_desc, keymap) = desc.to_wallet_descriptor(Network::Testnet).unwrap();
+        let signers_container = Arc::new(SignersContainer::from(keymap));
+        let policy = wallet_desc
+            .extract_policy(signers_container)
+            .unwrap()
+            .unwrap();
+
+        println!("desc policy = {:?}", policy); // TODO remove
+
+        // TODO how should this merge timelocks?
+
+        let (prvkey1, _pubkey1, _fingerprint1) = setup_keys(TPRV0_STR);
+        let locktime_seconds0 = 500000100;
+        let locktime_seconds1 = 500000200;
+        let desc = descriptor!(sh (and_v (+v pk prvkey1), (and_v (+v after locktime_seconds0), (after locktime_seconds1)))).unwrap();
+        let (wallet_desc, keymap) = desc.to_wallet_descriptor(Network::Testnet).unwrap();
+        let signers_container = Arc::new(SignersContainer::from(keymap));
+        let policy = wallet_desc
+            .extract_policy(signers_container)
+            .unwrap()
+            .unwrap();
+
+        println!("desc policy = {:?}", policy); // TODO remove
+
+        // TODO how should this merge timelocks?
+    }
+}
```

Comment:
I would comment out those tests that are supposed to fail but not failing yet, otherwise at a first glance it might look like those parts are already working since the tests are not asserting anything

Reply:
Good point, I've commented out the policy tests that had the TODOs.